src/core/server/saved_objects/migrations/actions/es_errors.ts

```diff
@@ -7,25 +7,31 @@
  */
 import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
 
-export const isWriteBlockException = ({ type, reason }: estypes.ErrorCause): boolean => {
+export const isWriteBlockException = (errorCause?: estypes.ErrorCause): boolean => {
   return (
-    type === 'cluster_block_exception' &&
-    reason.match(/index \[.+] blocked by: \[FORBIDDEN\/8\/.+ \(api\)\]/) !== null
+    typeof errorCause !== 'undefined' &&
+    errorCause.type === 'cluster_block_exception' &&
+    errorCause.reason.match(/index \[.+] blocked by: \[FORBIDDEN\/8\/.+ \(api\)\]/) !== null
   );
 };
 
-export const isIncompatibleMappingException = ({ type }: estypes.ErrorCause): boolean => {
-  return type === 'strict_dynamic_mapping_exception' || type === 'mapper_parsing_exception';
+export const isIncompatibleMappingException = (errorCause?: estypes.ErrorCause): boolean => {
+  return (
+    typeof errorCause !== 'undefined' &&
+    (errorCause.type === 'strict_dynamic_mapping_exception' ||
+      errorCause.type === 'mapper_parsing_exception')
+  );
 };
 
-export const isIndexNotFoundException = ({ type }: estypes.ErrorCause): boolean => {
-  return type === 'index_not_found_exception';
+export const isIndexNotFoundException = (errorCause?: estypes.ErrorCause): boolean => {
+  return typeof errorCause !== 'undefined' && errorCause.type === 'index_not_found_exception';
 };
 
-export const isClusterShardLimitExceeded = ({ type, reason }: estypes.ErrorCause): boolean => {
+export const isClusterShardLimitExceeded = (errorCause?: estypes.ErrorCause): boolean => {
   return (
-    type === 'validation_exception' &&
-    reason.match(
+    typeof errorCause !== 'undefined' &&
+    errorCause.type === 'validation_exception' &&
+    errorCause.reason.match(
       /this action would add .* shards, but this cluster currently has .* maximum normal shards open/
```

Comment:
NIT: 

```ts
return (
    errorCause?.type === 'validation_exception' &&
    errorCause?.reason.match(...)
```

Reply:
good point! for some reason, I initially got type errors when I first tried that option... it probably was something else... 🤷 

Changed!